File: practice21/9.cpp
//https://www.nowcoder.com/practice/bfd8234bb5e84be0b493656e390bdebf?tpId=37&tqId=21284&tPage=4&rp=&ru=/ta/huawei&qru=/ta/huawei/question-ranking
#include<iostream>
using namespace std;

int Put_numbers(int m, int n)
{
	if (m <= 0)
	{
		return 0;
	}
	if (m == 1 || n == 1)
	{
		return 1;
	}
	if (m == n)
	{
		return Put_numbers(m, n - 1) + 1;
	}
	else
	{
		return Put_numbers(m, n - 1) + Put_numbers(m - n, n);
	}
}

int main()
{
	int m, n;
	while (cin >> m >> n)
	{
		cout << Put_numbers(m, n) << endl;
	}
	system("pause");
	return 0;
}

File: practice28/9.java
//: strings/ReplacingStringTokenizer.java

//https://www.nowcoder.com/practice/b6edb5ca15d34b1eb42e4725a3c68eba?tpId=98&tqId=33062&tPage=12&rp=12&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);
        int x=scanner.nextInt();
        String str[]=new String[x];
        int []pt=new int[x];
        for(int i=0;i<x;i++){
            str[i]=scanner.next();
        }
        int m,n;
        for(int i=0;i<x;i++){
            for(m=0,n=str[i].length()-1;m<n;){
                if(str[i].charAt(m)!=str[i].charAt(n)){
                    if(str[i].charAt(m+1)==str[i].charAt(n)){
                        pt[i]=m;
                        break;
                    }
                    if(str[i].charAt(m)==str[i].charAt(n-1)){
                        pt[i]=n;
                        break;
                    }
                }
                else{
                    m++;
                    n--;
                }
            }
            if(m>=n){
                pt[i]=-1;
            }
        }
        for(int num:pt){
            System.out.println(num);
        }
    }
}
